More const

thalieht opened the pull request:
I don't think it helps repeating commit messages here so i won't.

I found many 'auto' vars that if i were to add const it would be `*const` (hovering over them in qtcreator says so). Would there be any value to that and if yes should i do it in the form of "auto X ---> X *const"?

Chocobo1 commented:
>Would there be any value to that and if yes should i do it in the form of "auto X ---> X *const"?

I don't think I understand, can you put a real example from our codebase?

thalieht commented:
>I don't think I understand, can you put a real example from our codebase?

e.g.
https://github.com/qbittorrent/qBittorrent/blob/bb041c0eca44bd5602cabfce01fe292f7b2713b9/src/base/http/server.cpp#L113
making it "const auto ..." and hovering over it shows it became "Http::connection *const"

thalieht commented:
>didn't think to explicitly add the asterisk which has the desired effect... i'll do that.

Turns out in most cases i can't do that so back to the question. Is there any value to making the pointer const?

Chocobo1 commented:
In the example, adding it `const` such as `const auto connection = ...` isn't any different from your commit "Add const to almost all remaining vars and arguments that qualify".

Chocobo1 commented:
"Use aggregate initialization on some default arguments" it should be "uniform initialization".

thalieht commented:
Added a few more const to the first commit, here's the diff if it helps https://gist.github.com/thalieht/4f6a0058ff48fff8a8916d9cdeeef797
and changed the other commit's msg to "uniform initialization"

glassez commented on src/gui/fspathedit_p.cpp:
```diff
@@ -251,7 +251,7 @@ void Private::FileLineEdit::keyPressEvent(QKeyEvent *e)
     }
 
-    auto *validator = qobject_cast<const FileSystemPathValidator *>(this->validator());
+    const auto *validator = qobject_cast<const FileSystemPathValidator *>(this->validator());
```
We discussed this line in previous PR. `auto` will give you the same result.

thalieht replied:
Yeah sorry, didn't even look what's on the right hand here.

thalieht replied:
Reverted.

thalieht commented on src/base/preferences.h:
```diff
@@ -321,9 +321,9 @@ class Preferences : public QObject
     void setPropFileListState(const QByteArray &state);
     int getPropCurTab() const;
-    void setPropCurTab(const int &tab);
+    void setPropCurTab(const int tab);
```
Oops, the order of commits made me overlook them. i'll remove both const and ref in 1st commit.

Chocobo1 commented:
👍
